src/basic/terminal-util.c

```diff
@@ -498,6 +498,15 @@ int release_terminal(void) {
         return r;
 }
 
+int terminal_new_session(void) {
+
+        if (!isatty_safe(STDOUT_FILENO) || !isatty_safe(STDERR_FILENO))
+                return -ENXIO;
```

Comment:
why this restriction? that seems unnecessary and potentially problematic?

it's entirely fine if a process has a ctty to which it has no open fds... i see no reason why we should check for this case here.